src/OrchardCore.Modules/OrchardCore.Email/Drivers/SmtpSettingsDisplayDriver.cs

```diff
@@ -39,7 +39,7 @@ public override async Task<IDisplayResult> EditAsync(SmtpSettings settings, Buil
         {
             var user = _httpContextAccessor.HttpContext?.User;
 
-            if (!await _authorizationService.AuthorizeAsync(user, Permissions.ManageEmailSettings))
+            if (user == null || !await _authorizationService.AuthorizeAsync(user, Permissions.ManageEmailSettings))
```

Comment:
Why the null check? Doesn't work previously?!!

Reply:
There should be no need I agree with @hishamco

Reply:
https://github.com/OrchardCMS/OrchardCore/blob/454105788cd2e9184c15a620eb3c212ad30cb488/src/OrchardCore/OrchardCore.Infrastructure/Security/AuthorizationServiceExtensions.cs#L17-L20